### PR TITLE
Configure dependency management for maven projects

### PR DESCRIPTION
**Pull request by zaharchenko-evgeny**
This feature is required for Jmix 1.0.x

### CODE REVIEW

File: src/main/groovy/io/jmix/build/JmixBuildPlugin.groovy [hunk 1]
```diff
@@ -265,4 +266,16 @@ class JmixBuildPlugin implements Plugin<Project> {
     private Set<Project> getJavaSubProjects(Project rootProject) {
         rootProject.subprojects.findAll { subproject -> subproject.plugins.hasPlugin(JavaPlugin) }
     }
+
+    private void setupDependencyManagement(Project project) {
+        if (!project.name.startsWith('sample')) {
+            project.with {
+                dependencies {
+                    api platform("org.springframework.boot:spring-boot-starter-parent:2.5.2")
+                    api platform("io.jmix.bom:jmix-bom:$bomVersion")
```
Comment (andreysubbotin) [inline src/main/groovy/io/jmix/build/JmixBuildPlugin.groovy]: еще одна зависимость на Spring Boot=) без нее не обойтись ли

File: src/main/groovy/io/jmix/build/JmixBuildPlugin.groovy [hunk 2]
```diff
@@ -265,4 +266,16 @@ class JmixBuildPlugin implements Plugin<Project> {
     private Set<Project> getJavaSubProjects(Project rootProject) {
         rootProject.subprojects.findAll { subproject -> subproject.plugins.hasPlugin(JavaPlugin) }
     }
+
+    private void setupDependencyManagement(Project project) {
+        if (!project.name.startsWith('sample')) {
+            project.with {
+                dependencies {
+                    api platform("org.springframework.boot:spring-boot-starter-parent:2.5.2")
```
Comment (knstvk) [inline src/main/groovy/io/jmix/build/JmixBuildPlugin.groovy]: Why not `org.springframework.boot:spring-boot-dependencies:2.5.2` which is used in BOM's build.gradle?

**Review by knstvk — COMMENTED** [review]
Please check if Maven works if you just add Jmix BOM to starters, e.g.:
```
group = 'io.jmix.core'
archivesBaseName = 'jmix-core-starter'

dependencies {
    api platform("io.jmix.bom:jmix-bom:$bomVersion")

    api 'io.jmix.core:jmix-core'

    implementation 'org.springframework.boot:spring-boot-autoconfigure'
}
```

**Comment by zaharchenko-evgeny** [comment]
Works fine with only Jmix bom dependency. I have updated PR

**Pull request merged**